src/components/App/App.js

```diff
@@ -18,21 +21,41 @@ function App() {
       .catch(error => setError(`Sorry, something went wrong. Please try again later.`))
       console.log(error)
   }, [])
+
+  const handleClick = () => {
+    getRandomQuote()
+      .then(data => {
+        setQuote({...data, isFavorite: false})
+      })
+  }
+
+  const handleFavorite = (event) => { 
+    const target = event.target.id;
+    console.log(typeof target)
+    if (quote._id === target && !quote.isFavorite) {
+      setQuote((prevQuote) => ({...prevQuote, isFavorite: true}))
+      setFavorites((prevFavorites) => [...prevFavorites, {...quote, isFavorite: true, _id: Date.now()}])
+    } else if (quote._id === target && quote.isFavorite) {
+      setAlreadySaved(true)
+    } else {
+      const filteredFavorites = favorites.filter(favorite => favorite._id !== parseInt(target))
+      setFavorites(filteredFavorites)
+    }
```

Comment:
Ideally this would be a simple if/ else. I can't see any use or current functionality to the `else if` part of this. I think that setting the current quote's isFavorite to true in the `if` check is all that you need to know as to whether it's been favorited.